File: TDDMicroExercises.YoursSolutions/Java-Saleem-Siddiqui/src/test/java/tddmicroexercises/tirepressuremonitoringsystem/AlarmTest.java
package tddmicroexercises.tirepressuremonitoringsystem;

import org.junit.Before;
import org.junit.Test;
import org.mockito.Mock;
import static org.junit.Assert.assertThat;
import static org.hamcrest.Matchers.is;
import static java.lang.Boolean.FALSE;
import static java.lang.Boolean.TRUE;
import static org.mockito.Mockito.*;
import static org.mockito.MockitoAnnotations.initMocks;

public class AlarmTest {
    @Mock private ISensor sensor;

    @Before
    public void setUp() {
	initMocks(this);
    }

    @Test
    public void canCreateAlarmWithGivenSensor() {
	new Alarm(sensor);
    }

    @Test
    public void aNewAlarmIsOff() {
        assertThat((new Alarm()).isAlarmOn(), is(FALSE));
    }

    @Test
    public void alarmIsOnWhenSensorValueBelowThreshold() {
	when(sensor.popNextPressurePsiValue()).thenReturn(0D);
	Alarm alarm = new Alarm(sensor);
	alarm.check();
	assertThat(alarm.isAlarmOn(), is(TRUE));
    }
}

File: TDDMicroExercises.YoursSolutions/CSharp-benwu/TirePressureMonitoringSystem.Tests/AlarmTest.cs
using System;
using System.Collections.Generic;
using System.Text;
using Moq;
using NUnit.Framework;

namespace TDDMicroExercises.TirePressureMonitoringSystem.Tests
{
    [TestFixture]
    public class AlarmTest
    {
        [Test]
        public void GIVEN_pressure_psi_value_is_16_WHEN_check_pressure_value_in_alarm_THEN_alarm_is_on()
        {
            // Given
            var sensor = new Mock<ISensor>();
            var alarm = new Alarm(sensor.Object);

            sensor
                .Setup(s => s.PopNextPressurePsiValue())
                .Returns(16.0);

            // When
            alarm.Check();

            // Then
            Assert.IsTrue(alarm.AlarmOn);
            sensor.Verify(s => s.PopNextPressurePsiValue());
        }

        [Test]
        public void GIVEN_pressure_psi_value_is_17_WHEN_check_pressure_value_in_alarm_THEN_alarm_is_off()
        {
            // Given
            var sensor = new Mock<ISensor>();
            var alarm = new Alarm(sensor.Object);

            sensor
                .Setup(s => s.PopNextPressurePsiValue())
                .Returns(17.0);

            // When
            alarm.Check();

            // Then
            Assert.IsFalse(alarm.AlarmOn);
            sensor.Verify(s => s.PopNextPressurePsiValue());
        }
    }
}
